### PR TITLE
Wide gamut and platform view integration test.

### PR DESCRIPTION
**Pull request by jonahwilliams**
In https://github.com/flutter/engine/pull/48190 I discovered that overlay surfaces were not constructed with wide gamut settings. This adds a test that will fail until this is fixed.

### CODE REVIEW

File: dev/devicelab/lib/tasks/integration_tests.dart [hunk 1]
```diff
@@ -166,7 +166,6 @@ TaskFunction createWideGamutTest() {
   return IntegrationTest(
     '${flutterDirectory.path}/dev/integration_tests/wide_gamut_test',
     'integration_test/app_test.dart',
-    createPlatforms: <String>['ios'],
```
Comment (jonahwilliams) [inline dev/devicelab/lib/tasks/integration_tests.dart:169]: Removed so I can check in the platform view

File: dev/integration_tests/flutter_gallery/ios/Runner.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -461,9 +464,13 @@
 			baseConfigurationReference = 9740EEB21CF90195004384FC /* Debug.xcconfig */;
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
+				DEVELOPMENT_TEAM = S8QB4VV633;
```
Comment (gaaclarke) [inline dev/integration_tests/flutter_gallery/ios/Runner.xcodeproj/project.pbxproj:467]: I'd just revert this file.  You don't want to check this in and the other stuff is just formatting.

File: dev/integration_tests/wide_gamut_test/ios/Flutter/AppFrameworkInfo.plist [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (gaaclarke) [inline dev/integration_tests/wide_gamut_test/ios/Flutter/AppFrameworkInfo.plist:1]: I don't think you need to add all these files.  `flutter create --no-overwrite` is used.

File: dev/integration_tests/wide_gamut_test/ios/Runner/DummyPlatformView.swift [hunk 4]
```diff
@@ -0,0 +1,73 @@
+//
+//  DummyPlatformView.swift
+//  Runner
+//
+//  Created by Jonah Williams on 11/21/23.
+//
```
Comment (gaaclarke) [inline dev/integration_tests/wide_gamut_test/ios/Runner/DummyPlatformView.swift:6]: Fix copyright

**Comment by jonahwilliams** [comment]
Running at ToT you will get a validation error:

image

File: dev/integration_tests/wide_gamut_test/ios/Runner/DummyPlatformView.swift [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            binaryMessenger: messenger)
+    }
+
+    public func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
+          return FlutterStandardMessageCodec.sharedInstance()
+    }
+}
+
+class SolidColorView: UIView {
+  private var _started: Bool = false;
```
Comment (gaaclarke) [inline dev/integration_tests/wide_gamut_test/ios/Runner/DummyPlatformView.swift:39]: Above you don't use the underscore prefix.  I don't think its required in the swift style.

File: dev/integration_tests/wide_gamut_test/ios/Runner/DummyPlatformView.swift [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    setupView()
+  }
+
+  required init?(coder aDecoder: NSCoder) {
+    super.init(coder: aDecoder)
+    setupView()
+  }
+
+  private func setupView() {
+    backgroundColor = .blue
```
Comment (gaaclarke) [inline dev/integration_tests/wide_gamut_test/ios/Runner/DummyPlatformView.swift:52]: In order to duplicate the problem, shouldn't the wide gamut color come from the platform view?

**Review by gaaclarke — COMMENTED** [review]
Nice, thanks for the quick followup.  Here are a couple of questions/comments.

File: dev/integration_tests/wide_gamut_test/ios/Runner/DummyPlatformView.swift (hunk 6 above)
Comment (jonahwilliams) [reply dev/integration_tests/wide_gamut_test/ios/Runner/DummyPlatformView.swift:52]: No, the problem is the overlay flutter view

File: dev/integration_tests/wide_gamut_test/ios/Flutter/AppFrameworkInfo.plist (hunk 3 above)
Comment (jonahwilliams) [reply dev/integration_tests/wide_gamut_test/ios/Flutter/AppFrameworkInfo.plist]: Removed

File: dev/integration_tests/wide_gamut_test/ios/Runner/DummyPlatformView.swift (hunk 4 above)
Comment (jonahwilliams) [reply dev/integration_tests/wide_gamut_test/ios/Runner/DummyPlatformView.swift]: Done

File: dev/integration_tests/wide_gamut_test/ios/Runner/DummyPlatformView.swift (hunk 5 above)
Comment (jonahwilliams) [reply dev/integration_tests/wide_gamut_test/ios/Runner/DummyPlatformView.swift]: Done

File: dev/integration_tests/flutter_gallery/ios/Runner.xcodeproj/project.pbxproj (hunk 2 above)
Comment (jonahwilliams) [reply dev/integration_tests/flutter_gallery/ios/Runner.xcodeproj/project.pbxproj:467]: Done

File: dev/integration_tests/wide_gamut_test/ios/Runner/AppDelegate.swift [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+@UIApplicationMain
+@objc class AppDelegate: FlutterAppDelegate {
+  override func application(
+    _ application: UIApplication,
+    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
+  ) -> Bool {
+    GeneratedPluginRegistrant.register(with: self)
+
+      weak var registrar = self.registrar(forPlugin: "plugin-name")
```
Comment (gaaclarke) [inline dev/integration_tests/wide_gamut_test/ios/Runner/AppDelegate.swift:16]: I don't think you need `weak` here.

File: dev/integration_tests/wide_gamut_test/ios/Runner/DummyPlatformView.swift [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+  }
+}
+
+
+class FLNativeView: NSObject, FlutterPlatformView {
+    private var _view: UIView
```
Comment (gaaclarke) [inline dev/integration_tests/wide_gamut_test/ios/Runner/DummyPlatformView.swift:53]: Same note here about the underscore.

**Review by gaaclarke — APPROVED** [review]
LGTM modulo 2 swift notes

**Comment by jonahwilliams** [comment]
It looks like I reverted too many files

**Pull request merged**